src/Microsoft.AspNet.Mvc.ModelBinding/ValueProviders/FormValueProviderFactory.cs

```diff
@@ -19,8 +19,8 @@ public IValueProvider GetValueProvider([NotNull] ValueProviderFactoryContext con
             if (IsSupportedContentType(request))
             {
                 var culture = GetCultureInfo(request);
-                return new ReadableStringCollectionValueProvider<IFormDataValueProviderMetadata>(
-                    () => request.GetFormAsync(),
+                return new FormCollectionValueProvider(
+                    () => request.ReadFormAsync(),
```

Comment:
extra lambda isn't needed here, you should be able to pass `request.ReadFormAsync`